lake/seekindex/writer.go

```diff
@@ -22,22 +24,31 @@ func NewWriter(w zio.Writer) *Writer {
 	}
 }
 
-func (w *Writer) Write(key zed.Value, offset int64) error {
+func (w *Writer) Write(key zed.Value, number uint64, offset int64) error {
 	b := w.builder
 	b.Reset()
 	if zed.IsContainerType(key.Type) {
 		b.AppendContainer(key.Bytes)
 	} else {
 		b.AppendPrimitive(key.Bytes)
 	}
+	b.AppendPrimitive(zed.EncodeUint(number))
 	b.AppendPrimitive(zed.EncodeInt(offset))
 	if w.typ != key.Type {
 		var schema = []zed.Column{
 			{"key", key.Type},
+			{"record", zed.TypeUint64},
 			{"offset", zed.TypeInt64},
 		}
 		w.recType = w.zctx.MustLookupTypeRecord(schema)
 		w.typ = key.Type
 	}
 	return w.writer.Write(zed.NewValue(w.recType, b.Bytes()))
 }
+
+func (w *Writer) Close() error {
+	if closer, ok := w.writer.(io.Closer); ok {
+		return closer.Close()
+	}
+	return nil
+}
```

Comment:
This doesn't look necessary. It also has foot-shooting potential because it makes it easy to call NewWriter with an io.WriterCloser wrapped by something returning an io.Writer and hiding io.WriterCloser.Close.

Reply:
Fair.